jwst/wavecorr/wavecorr.py

```diff
@@ -102,7 +102,7 @@ def apply_zero_point_correction(slit, reffile):
                                                  aperture_name, dispersion)
     ## TODO: set a DQ flag to a TBD value for pixels where dq_lam == 0.
     ## The only purpose of dq_lam is to set that flag.
-    slit.wavelength = lam - corr
+    slit.wavelength = lam - corr * 10 ** 6
```

Comment:
A comment here would be useful, to indicate the purpose of the scaling (e.g. converting meters to microns)